### ISSUE TITLE
MissingInternetConnectionException is appeared when it shouldn't be 

### ISSUE THREAD

**Issue by duprass** [WalletConnect/WalletConnectKotlinV2#270]
**Describe the bug**
MissingInternetConnectionException throws when network connection is available. 

**SDK Version**
 - Client: Kotlin
 - Version: 2.0.0-rc.0

**To Reproduce**
Error can occur during request or when just any Pairing exists 
```
com.walletconnect.sign.core.exceptions.client.WalletConnectException$MissingInternetConnectionException: No connection available
        at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.checkConnectionWorking$sign_release(RelayerInteractor.kt:85)
        at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.publishJsonRpcRequests$sign_release(RelayerInteractor.kt:96)
        at com.walletconnect.sign.engine.domain.SignEngine$updateSession$2.invoke(SignEngine.kt:243)
        at com.walletconnect.sign.engine.domain.SignEngine$updateSession$2.invoke(SignEngine.kt:242)
        at com.walletconnect.sign.storage.sequence.SequenceStorageRepository$insertUnAckNamespaces$1$3$4.invoke(SequenceStorageRepository.kt:205)
        at com.walletconnect.sign.storage.sequence.SequenceStorageRepository$insertUnAckNamespaces$1$3$4.invoke(SequenceStorageRepository.kt:205)
        at com.squareup.sqldelight.TransacterImpl.transactionWithWrapper(Transacter.kt:260)
        at com.squareup.sqldelight.TransacterImpl.transaction(Transacter.kt:214)
        at com.squareup.sqldelight.Transacter$DefaultImpls.transaction$default(Transacter.kt:72)
        at com.walletconnect.sign.storage.sequence.SequenceStorageRepository.insertUnAckNamespaces(SequenceStorageRepository.kt:181)
        at com.walletconnect.sign.engine.domain.SignEngine.updateSession$sign_release(SignEngine.kt:242)
        at com.walletconnect.sign.client.SignProtocol.update(SignProtocol.kt:184)
        at com.walletconnect.sign.client.SignClient.update(Unknown Source:12)
```

**Expected behavior**
Error should be thrown when there's indeed no internet connection, and not when it's alive actually. 

**Device (please complete the following information):**
The error is repeatable on any Android device

**Comment by Elyniss** [comment]
Hey @duprass! Thanks for submitting the bug report. May I ask for filling more info about device you are using?

**Device (please complete the following information):**
 - Device: [e.g. Pixel]
 - OS: [e.g. Android 12]
 - Browser [e.g. stock browser, chrome]

Also, may I ask about frequency of this error?

**Comment by duprass** [comment]
Hello @Elyniss!
As it was written it's reproduced on Android OS starting from 24 version regardless on device. The browsers were used on desktop as chrome, safari, brave. It seems that it doesn't depends on it too.

**Comment by Elyniss** [comment]
I spent the last hour trying to reproduce this error with no success. 

What I did:
Using our sample Android wallet app and sample browser dapp, I created multiple paring and sessions with a physical Android device (Vivo V2109, Android 11) and desktop browser. After that, I called session requests from desktop Dapp, all of which I received on my phone. Later I emitted events from the Android device, and all of them were successful as well. I also left the device for a couple of minutes and retried the above tests. 

@duprass May I ask for guidance on how to reproduce it?

**Comment by mrbipbip** [comment]
Hello @Elyniss 

I can almost shared the same stacktrace but with the call of `SignClient.connet`, for a really high frequency (8/10 times)

```
com.walletconnect.sign.core.exceptions.client.WalletConnectException$MissingInternetConnectionException: No connection available
        at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.checkConnectionWorking$sign_release(RelayerInteractor.kt:85)
        at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.subscribe$sign_release(RelayerInteractor.kt:174)
        at com.walletconnect.sign.engine.domain.SignEngine.proposePairing(SignEngine.kt:125)
        at com.walletconnect.sign.engine.domain.SignEngine.proposeSequence$sign_release(SignEngine.kt:109)
        at com.walletconnect.sign.client.SignProtocol.connect(SignProtocol.kt:110)
        at com.walletconnect.sign.client.SignClient.connect(Unknown Source:17)
```

Of course, `SignClient.initialize` is called on my Application class, and other methods are called after.
I noticed that this error seems to be random for me, on an Android Emulator & on my personal phone (Pixel 4XL).

Sometimes, I can also encountered this issues from your dApps sample:
```
WalletConnectViewModel: com.walletconnect.sign.core.exceptions.client.WalletConnectException$MissingInternetConnectionException: No connection available
        at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.checkConnectionWorking$sign_debug(RelayerInteractor.kt:85)
        at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.subscribe$sign_debug(RelayerInteractor.kt:174)
        at com.walletconnect.sign.engine.domain.SignEngine.proposePairing(SignEngine.kt:125)
        at com.walletconnect.sign.engine.domain.SignEngine.proposeSequence$sign_debug(SignEngine.kt:109)
        at com.walletconnect.sign.client.SignProtocol.connect(SignProtocol.kt:110)
        at com.walletconnect.sign.client.SignClient.connect(Unknown Source:17)
        at com.walletconnect.dapp.ui.connect.ConnectViewModel.connectToWallet(ConnectViewModel.kt:79)
        at com.walletconnect.dapp.ui.connect.ConnectViewModel.connectToWallet$default(ConnectViewModel.kt:54)
        at com.walletconnect.dapp.ui.connect.pairing_generate.PairingGenerationDialogFragment.onViewCreated(PairingGenerationDialogFragment.kt:30)
```

I got some `timeout` error too, but i can't reproduce at this time.

Feel free to ask more detail if you need.

**Comment by MagsMagnoli** [comment]
I am also seeing this issue @Elyniss 

Device: Pixel 6 Pro
OS: 12

I'm creating a Dapp app and initialize in the application class. On click of a button I am calling `SignClient.connect` and I receive the following crash

```
com.walletconnect.sign.core.exceptions.client.WalletConnectException$MissingInternetConnectionException: No connection available
         at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.checkConnectionWorking$sign_release(RelayerInteractor.kt:85)
         at com.walletconnect.sign.json_rpc.domain.RelayerInteractor.subscribe$sign_release(RelayerInteractor.kt:174)
         at com.walletconnect.sign.engine.domain.SignEngine.proposePairing(SignEngine.kt:125)
         at com.walletconnect.sign.engine.domain.SignEngine.proposeSequence$sign_release(SignEngine.kt:109)
         at com.walletconnect.sign.client.SignProtocol.connect(SignProtocol.kt:110)
         at com.walletconnect.sign.client.SignClient.connect(Unknown Source:17)
         at com.myapp.android.wallet.walletconnect.WalletConnectGenerateDeepLinkUriOperation.invoke(WalletConnectDapp.kt:54)
         at com.myapp.android.core.wallet.GeneratePairingDeepLinkUriUseCase.execute(MyFile.kt:12)
         at com.myapp.android.auth.ui.AuthScreenKt$AuthScreen$1$1$1.invokeSuspend(MyScreen.kt:55)
         at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
         at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
         at androidx.compose.ui.platform.AndroidUiDispatcher.performTrampolineDispatch(AndroidUiDispatcher.android.kt:81)
         at androidx.compose.ui.platform.AndroidUiDispatcher.access$performTrampolineDispatch(AndroidUiDispatcher.android.kt:41)
         at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.run(AndroidUiDispatcher.android.kt:57)
         at android.os.Handler.handleCallback(Handler.java:938)
         at android.os.Handler.dispatchMessage(Handler.java:99)
         at android.os.Looper.loopOnce(Looper.java:201)
         at android.os.Looper.loop(Looper.java:288)
         at android.app.ActivityThread.main(ActivityThread.java:7870)
         at java.lang.reflect.Method.invoke(Native Method)
         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by Elyniss** [comment]
Should be fixed with #305
